awkward/array/jagged.py

```diff
@@ -1350,9 +1362,42 @@ def concatenate(isclassmethod, cls_or_self, arrays):
 
         return cls(starts, stops, content)
 
-    @classmethod
-    def stack(cls, first, *rest):    # each item in first followed by second, etc.
-        raise NotImplementedError
+    @awkward.util.bothmethod
+    def stack(isclassmethod, cls_or_self, arrays):    # each item in first followed by second, etc.
+
+        if isclassmethod: 
+            cls = cls_or_self
+            if not all(isinstance(x, JaggedArray) for x in arrays):
+                raise TypeError("cannot stack non-JaggedArrays with JaggedArray.stack")
+        else:
+            self = cls_or_self
+            cls = self.__class__
+            if not isinstance(self, JaggedArray) or not all(isinstance(x, JaggedArray) for x in arrays):
+                raise TypeError("cannot stack non-JaggedArrays with JaggedArray.stack")
+            arrays = (self,) + tuple(arrays)
+
+        if len(set([len(a) for a in arrays])) > 1:
+            raise ValueError("cannot stack JaggedArrays of different lengths")
```

Comment:
What about when `len(arrays) == 0`? Maybe handle both error states without creating a `set`:

```python
if len(arrays) == 0:
    raise ValueError("at least one array must be provided")   # this can only happen in the classmethod case
if any(len(x) != len(arrays[0]) for x in arrays):
    raise ValueError("cannot concatenate JaggedArrays of different lengths with axis=1")
```